### PR TITLE
feat: Kraken WS Provider

### PR DESCRIPTION
**Pull request by davidterpay**

### CODE REVIEW

**Comment by aljo242** [comment]
misspelled word !

**Comment by davidterpay** [comment]
still need to add testing

File: providers/websockets/kraken/messages.go [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+//	  	 	],
+//	   	 	"v": [				// Volume array
+//	    	  "2634.11501494", 	// Value Today
+//	    	  "3591.17907851"	// Value 24h
+//	   	 	]
+//	 	 },
+//	  	"ticker", 				// Channel name
+//	  	"XBT/USD" 				// Asset pair
+//
+// ]
```
Comment (davidterpay) [inline providers/websockets/kraken/messages.go:286]: formatting is ok when look at github.dev not sure why it looks incorrect on the other.

File: providers/base/websocket/handlers/ws_data_handler.go [hunk 2]
```diff
@@ -14,8 +14,8 @@ import (
 type WebSocketDataHandler[K comparable, V any] interface {
 	// HandleMessage is used to handle a message received from the data provider. Message parsing
 	// and response creation should be handled by this data handler. Given a message from the web socket
-	// the handler should either return a response or an update message.
-	HandleMessage(message []byte) (response providertypes.GetResponse[K, V], updateMessage []byte, err error)
+	// the handler should either return a response or a set of update messages.
+	HandleMessage(message []byte) (response providertypes.GetResponse[K, V], updateMessages []WebsocketEncodedMessage, err error)
```
Comment (davidterpay) [inline providers/base/websocket/handlers/ws_data_handler.go:18]: adding this mostly for completeness with the other function signature

File: providers/websockets/kraken/messages.go (hunk 1 above)
Comment (davidterpay) [reply providers/websockets/kraken/messages.go:286]: also looks fine when looking locally

**Review by aljo242 — APPROVED** [review]

**Pull request merged**